solver/src/test/java/org/chocosolver/solver/constraints/binary/SquareTest.java

```diff
@@ -43,8 +43,8 @@ public void testBigBoundBound() {
         solver.setSearch(Search.randomSearch(ArrayUtils.append(x,x2), 0));
         while (solver.solve()) ;
         assertEquals(solver.getSolutionCount(), 184320);
-        assertEquals(solver.getFailCount(), 412230);
-        assertEquals(solver.getNodeCount(), 780869);
+        assertEquals(solver.getFailCount(), 270905);
```

Comment:
why does the number of nodes change ?
which filtering rule was missing ? 
(It could also be that the new filtering is weaker and the search more lucky)

Reply:
It was in method updateLowerBoundofX(). Whenever both variables had bounded domains and Y.getLB() < 0 < Y.getUB(), then we would not filter the lower of X even if it didn't correspond to a perfectNth value. In the case of square values, let say that X in [7,25] and Y in [-5,5]. With former code, X's lower bound would not have been updated to 9, whereas it is the case now.

Reply:
shouldn't we have a similar treatment in updateUpperBoundofX() ?

Reply:
No, the case doesn't happen in updateUpperBoundofX() as we explicitly use the function pow(int n) with the bounds of Y. And since we loop on updating the bounds of variables X and Y, the case doesn't happen for X.UB, which is well updated wrt Y's bounds.